### ISSUE TITLE
[pytorch] [Feature Request] SoftArgMax Function for differentiable argmax

### ISSUE THREAD

**Issue by Naman-ntc** [pytorch/pytorch#7766]
Softargmax is used quite many place. It can be used as a differentiable alternative to argmax function and thus comes handy at times! [A brief idea](https://bouthilx.wordpress.com/2013/04/21/a-soft-argmax/) about softargmax function.

What do you think about adding it into the nn Module?

I used `nn.Softmax` as well as `torch.arange` to implement it [here on forums](https://discuss.pytorch.org/t/softargmax-in-pytorch-in-2d/17931).

I can create a PR if you just mention relevant files I need to play around for the function.

**Comment by SsnL** [comment]
please don't create new issues to "bump" your request...

**Comment by Naman-ntc** [comment]
Didnt mean to catch attention :sweat_smile: , I didnt write old issue clearly so closed and reopened so that could be clear. Sorry!

**Comment by SsnL** [comment]
You can just use softmax and dot product with indices.

**Comment by bhack** [comment]
Check https://github.com/MWPainter/cvpr2019/blob/master/stitched/soft_argmax.py#L117. /cc @MWPainter if he want to contribute with a PR to the pytroch repo.

**Comment by YohanObadia** [comment]
> 
> 
> You can just use softmax and dot product with indices. I don't think we should include this in core.

Not if you want to emphasize more the coordinates of the accurate argmax with a beta value as such : $\sum_i{\frac{i\times\exp{\beta x_i}}{\sum_j{\exp{\beta x_j}}}}$
